### PR TITLE
builder,provisioner: add kubernetes builder

### PR DESCRIPTION
**Pull request by guilhermebr**

### CODE REVIEW

File: provision/kubernetes/deploy.go [hunk 1]
```diff
@@ -224,6 +257,7 @@ func createBuildPod(params buildPodParams) error {
 		fmt.Fprintln(params.attachOutput, " ---> Cleaning up")
 	}
 	return waitForPod(params.client, pod.Name, false, kubeConf.PodReadyTimeout)
+
```
Comment (cezarsa) [inline provision/kubernetes/deploy.go]: Empty line 😛

File: provision/kubernetes/provisioner.go [hunk 2]
```diff
@@ -616,6 +618,46 @@ func (p *kubernetesProvisioner) UpdateNode(opts provision.UpdateNodeOptions) err
...
+		writer: evt,
+	}
+	err = servicecommon.RunServicePipeline(manager, a, newImage, nil)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	return newImage, nil
+}
+
 func (p *kubernetesProvisioner) ImageDeploy(a provision.App, imageID string, evt *event.Event) (string, error) {
```
Comment (cezarsa) [inline provision/kubernetes/provisioner.go]: Can't we remove this method (along with `UploadDeploy`) now as it'll no longer be used?

Comment (guilhermebr) [reply provision/kubernetes/provisioner.go]: Sure! I'm just moving the tests to builder package and I will remove both methods from provider in a next commit

**Comment by gfleury** [comment]
+1, any ETA?

**Comment by guilhermebr** [comment]
@gfleury I'm coming back to work on this

File: app/image/image.go [hunk 3]
```diff
@@ -515,6 +515,27 @@ func GetAppImageBySuffix(appName, imageIdSuffix string) (string, error) {
 	return "", &InvalidVersionErr{Image: inputImage}
 }
 
+func SplitImageName(imageName string) (repo, tag string) {
```
[the remark this answers is not in the document]
Comment (guilhermebr) [reply app/image/image.go]: I actually moved the function to the image package and forgot to remove it from docker

File: builder/kubernetes/builder.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	builder.Register("kubernetes", &kubernetesBuilder{})
+}
+
+func (b *kubernetesBuilder) Build(prov provision.BuilderDeploy, app provision.App, evt *event.Event, opts *builder.BuildOpts) (string, error) {
+	p, ok := prov.(provision.BuilderDeployKubeClient)
+	if !ok {
+		return "", errors.New("provisioner not supported")
+	}
+	if opts.BuildFromFile {
+		return "", errors.New("build image from Dockerfile is not yet supported")
```
[the remark this answers is not in the document]
Comment (guilhermebr) [reply builder/kubernetes/builder.go]: I''m not sure... it's two different situations. 
BuildFromFile is not supported yet by Tsuru and ArchiveURL is not supported for now only by kubernetes Builder

**Pull request merged**